lib/rails_response_dumper/runner.rb

```diff
@@ -60,5 +48,24 @@ def rollback_after
         yield
       end
     end
+
+    def handle_repsonses(responses, dump_block, dumper_dir)
+      responses.each_with_index do |response, index|
+        if response.status == dump_block.expected_status_code
+          mime = response.content_type.split(/ *; */).first
+          extension = MIME::Types[mime].first.preferred_extension
+          File.write("#{dumper_dir}/#{index}.#{extension}", response.body)
+        elsif response.status == 204
```

Comment:
I don't think we should treat 204 as a special case. Instead, we should allow for different responses within the same dump block to have different status codes. Whether they are 204, 302, or 404, or whatever.

Doing so would solve the general problem, not just one surfaced issue.